types/src/state_store/state_key.rs

```diff
@@ -26,36 +24,42 @@ pub enum StateKey {
     Raw(Vec<u8>),
 }
 
+#[derive(ToPrimitive)]
+enum StateKeyPrefix {
+    AccountAddress,
+    AccessPath,
+    Raw = 256,
+}
+
+impl StateKeyPrefix {
+    fn to_bytes(&self) -> Vec<u8> {
+        let byte = self
+            .to_u8()
+            .expect("Failed to convert StateKeyPrefix to u8");
+        vec![byte]
+    }
+}
+
 struct RawStateKey {
     bytes: Vec<u8>,
 }
 
 impl From<&StateKey> for RawStateKey {
     fn from(key: &StateKey) -> Self {
-        match key {
+        let (prefix, raw_key) = match key {
             StateKey::AccountAddressKey(account_address) => {
-                let mut account_address_prefix = ACCOUNT_ADDRESS_KEY_PREFIX.as_bytes().to_vec();
-                account_address_prefix.extend(account_address.to_vec());
-                RawStateKey {
-                    bytes: account_address_prefix,
-                }
+                (StateKeyPrefix::AccountAddress, account_address.to_vec())
             }
             StateKey::AccessPath(access_path) => {
-                let mut account_access_path_prefix =
-                    ACCOUNT_ACCESS_PATH_KEY_PREFIX.as_bytes().to_vec();
-                account_access_path_prefix.extend(access_path.address.to_vec());
-                RawStateKey {
-                    bytes: account_access_path_prefix,
-                }
+                // TODO: include the path part
```

Comment:
@sitalkedia  was it deliberate that this doesn't include the `path` part?

Reply:
@msmouse - This was an oversight. As of now, this path is not exercised so this is not an issue, but let's include the prefix in this diff if you are okay.

Reply:
Actually, it seems like it was included earlier. Any reason you choose to remove it?

Reply:
it had ap.address, but not ap.path?

Reply:
Ah, right. This is an oversight.